lib/gtfs/cache_file.ex

```diff
@@ -0,0 +1,118 @@
+defmodule Gtfs.CacheFile do
+  @moduledoc """
+  This module holds the logic for loading and saving cache files
+  in dev and test environments with the goal of decreasing startup
+  time so devs can work more quickly.
+
+  If GTFS data changes, the cache files should be deleted/removed
+  to allowing caching of the changes.
+  """
+  require Logger
+
+  @directory Path.join([File.cwd!(), "priv/gtfs_cache"])
+
+  @doc """
+  The application should use the file to load state for GTFS
+  if the the envs are :dev or :test.
+
+    iex> Mix.env
+    :test
+    iex> CacheFile.should_use_file?
+    true
+  """
+  @spec should_use_file?() :: boolean
+  def should_use_file? do
+    Application.get_env(:skate, __MODULE__)[:cache_filename] != nil
+  end
+
+  @doc """
+  Attempt to load a cache file.
+
+  This will attempt to load a cache file in dev and test. It will fail
+  if the Mix.env is not dev or test, it will fail if the file does not exist
+  and it will fail if the loaded term is not a map (minimal validation).
+
+  The validation for loading this file must be appended in later code. No
+  such validator currently exists.
+  """
+  @spec load_gtfs() :: {:ok, Gtfs.t()} | {:error, atom}
+  def load_gtfs() do
+    Logger.info(fn -> "Loading gtfs cache from default path" end)
+
+    cache_filename()
+    |> generate_filepath()
+    |> load_gtfs()
+  end
+
+  @spec load_gtfs(String.t()) :: {:ok, Gtfs.t()} | {:error, atom}
+  def load_gtfs(filepath) when is_binary(filepath) do
+    Logger.info(fn -> "Loading gtfs cache from file #{filepath}" end)
+
+    with {:ok, binary_cache} <- File.read(filepath),
+         {:ok, state} when is_map(state) <- binary_to_term(binary_cache) do
+      Logger.info(fn -> "Loaded gtfs cache from file #{filepath}" end)
+      {:ok, state}
+    else
+      _ ->
+        Logger.info(fn -> "Failed to load gtfs cache from file #{filepath}" end)
+        {:error, :cache_not_loaded}
+    end
+  end
+
+  def load_gtfs(_) do
```

Comment:
I'm having a hard time following where this case comes from. Is it that if we're on prod, the `cache_filename()` is `nil`? But `generate_filepath()` doesn't have a case for that.

Instead of passing nils around, I'd prefer to notice the `nil` at a higher level. Something like
```elixir
def load_gtfs() do
  case cache_filename() do
    nil -> {:error, "no cache file configured"}
    filename ->
      load_gtfs(generate_filepath(filename)) # we know the path exists now, so load_gtfs/1 doesn't need a case for nil.
```

Reply:
Removed this case, there wasn't a specific intention behind it.